driver-core/src/main/com/mongodb/TaggableReadPreference.java

```diff
@@ -293,9 +295,6 @@ private ServerDescription findMostUpToDateSecondary(final ClusterDescription clu
                 }
             }
         }
-        if (mostUpdateToDateSecondary == null) {
-            throw new MongoInternalException("Expected at least one secondary in cluster description: " + clusterDescription);
-        }
         return mostUpdateToDateSecondary;
```

Comment:
The test `NoKnownServers.json` was returning the above exception when no error is expected for the test. If no secondary is found (as seen in the test), returning `null` will signal there are no suitable servers.

Reply:
OK, I see.  So there is a behavior change.  That case used to throw an exception without waiting for server selection timeout.  Now it waits for server selection timeout to throw the error.